### ISSUE TITLE
Action Text only usable with integer PKs

### ISSUE THREAD

**Issue by ekampp** [rails/rails#39911]
### Steps to reproduce

1. Follow the [ActionText guide](https://edgeguides.rubyonrails.org/action_text_overview.html) and then using the [`with_rich_text_<column>` method](https://edgeguides.rubyonrails.org/action_text_overview.html#avoid-n-1-queries).

### Expected behavior

This should work with non-integer primary keys (e.g. UUIDs)

### Actual behavior

It does not work with UUID primary keys and raises this exception:

```
ActiveRecord::StatementInvalid (PG::UndefinedFunction: ERROR:  operator does not exist: bigint = uuid
LINE 1: ...me" = $2 AND "action_text_rich_texts"."record_id" = "table...
                                                             ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
):
```

### System configuration
**Rails version**: 6.0.3.2

**Ruby version**: 2.7.1

**Comment by Korkey128k** [comment]
Possibly related to #40010. Can you provide your full install procedure so I can test?

Thanks!

**Comment by ekampp** [comment]
Yeah, it probably is, @Korkey128k. I pretty much did the exact steps you refer to in #40010, except my problems happen with action text, but it seems that the problem is in fact an incorrect foreign key column type because a bunch of the Rails internals and generators don't respect non-integer PKs yet.

**Comment by Korkey128k** [comment]
@ekampp Yep, fix is here #37583, slated for `6.1.0`

**Comment by stevenbuccini** [comment]
Hey everyone, I think this is something different than what has previously been fixed or there has been a regression. I am running off `rails@master` and so I have pulled in #37583 and have the appropriate config set. 

Despite this, the generated migrations run as a part of the `bin/rails action_text:install` command did not include the appropriate `type: uuid` option. My problem manifested itself as described in [this blog post](https://blog.natetodd.com/action/).
